akka-docs/src/main/paradox/stream/operators/Source-or-Flow/zipAll.md

```diff
@@ -0,0 +1,35 @@
+# zipAll
+
+Combines all elements from each of multiple sources into @scala[tuples] @java[*Pair*] and passes the @scala[tuples] @java[pairs] downstream.
+
+@ref[Fan-in operators](../index.md#fan-in-operators)
+
+@@@div { .group-scala }
+
+## Signature
+
+@@signature [Flow.scala](/akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala) { #zipAll }
+
+@@@
+
+## Description
+
+Combines all elements from each of multiple sources into @scala[tuples] @java[*Pair*] and passes the @scala[tuples] @java[pairs] downstream.
+
+
+@@@div { .callout }
+
+**emits** when all of the inputs have an element available, once part (but not all) of the inputs complete their values are substituted by the provided defaults.
```

Comment:
I found this sentence a bit hard to parse, what about going for just  "when all of the inputs have an element available" and explaining about emitting the default elements in the description text instead?

Reply:
Hi @johanandren , I think there's no escaping allocating the Tuple2 instances. actually the underlying zip operator behaves exactly the same (unless you consider the extra allocation n the map stage). in any case I think 'user space' operators are much simpler and easier to understand than custom stages.

looking at the code again I'd like to eliminate code duplication between zipAll and zipAllMat, I'll try to craft something and push a commit for that.

re. 'emits' documentation, what about:
at first emits when both inputs emit, and then as long as any input emits (coupled to the default value of the completed input).

in any case, this doc string has to be modified 6 (!) places (1 scala api, 4 java apis, documentation), right?

Reply:
Yes that sounds like a good text to me.